wire-ios-data-model/Source/MLS/Migration/ProteusToMLSMigrationCoordinator.swift

```diff
@@ -117,6 +125,44 @@ public class ProteusToMLSMigrationCoordinator: ProteusToMLSMigrationCoordinating
         }
     }
 
+    func finalizeMigrationIfNeeded() async throws {
+        let mixedConversations = [ZMConversation]()
+        if migrationForceTimeHasArrived {
+            for conversation in mixedConversations {
+                await finalizeConversationMigration(conversation: conversation)
+            }
+        } else {
+            // 1. Sync users with the backend
+            try await syncUsersWithTheBackend()
+            // 2. Loop through conversations
+            for conversation in mixedConversations {
+                await finalizeConversationMigration(conversation: conversation)
+            }
```

Comment:
```
if !migrationForceTimeHasArrived {
    try await syncUsersWithTheBackend()
}
mixedConversations.map { await finalizeConversationMigration(conversation: $0) }
```

Reply:
I can't use forEach because it's not async.

`Cannot pass function of type '(ZMConversation) async -> ()' to parameter expecting synchronous function type`. I'm using the regular for loop instead.